packages/frontend/components/primitives/search/component.tsx

```diff
@@ -0,0 +1,63 @@
+import { Routes } from '@enums';
+import { useAppDispatch, useTypedSelector } from '@hooks';
+import { useRouter } from 'next/router';
+import { useEffect, useState } from 'react';
+import { actionSearch } from 'store/product';
+import { SearchInput } from '../input';
+import { Dropdown } from '../menu-dropdown';
+import * as styles from './styles';
+import type { SearchProps } from './types';
+
+const Search = ({ value, setValue, ...props }: SearchProps) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const dispatch = useAppDispatch();
+  const { searchedProducts } = useTypedSelector((state) => state.product);
+  const { push } = useRouter();
+
+  const callback = async ({ target }) => {
+    setValue(target.value);
+    await dispatch(actionSearch(target.value.toString()));
```

Comment:
maybe we can add debounce or throttle function here to send a search request with some interval but not on every onChange action, for example every 100ms or other value that will fit